build.rs

```diff
@@ -0,0 +1,20 @@
+fn main() {
+    let cfg = match autocfg::AutoCfg::new() {
+        Ok(cfg) => cfg,
+        Err(e) => {
+            println!("cargo:warning=async-io: failed to detect compiler features: {e}");
+            return;
+        }
+    };
+
+    // emit expected config settings; may be replaced with
+    // autocfg::emit_possibility if we update to autocfg>=1.5
+    println!("cargo:rustc-check-cfg=cfg(async_io_no_pipe)");
```

Comment:
This can be moved to `Cargo.toml`, right?

Reply:
Yep, I am happy to implement it there if preferred.